pygmo/test.py

```diff
@@ -844,6 +844,42 @@ def runTest(self):
         udp.p_distance(population(udp, 20))
 
 
+class minlp_rastrigin_test_case(_ut.TestCase):
+    """Test case for the MINLP Rastrigin
+
+    """
+
+    def runTest(self):
+        from .core import minlp_rastrigin, problem, population
+        udp = minlp_rastrigin(dim_c=2, dim_i=3)
+        prob = problem(udp)
+        self.assertTrue(prob.get_nx() == 5)
+        self.assertTrue(prob.get_nix() == 3)
+        self.assertTrue(prob.get_ncx() == 2)
+        pop = population(udp, 1)
+        self.assertTrue(int(pop.get_x()[0][-1]) == pop.get_x()[0][-1])
+        self.assertTrue(int(pop.get_x()[0][-2]) == pop.get_x()[0][-2])
+        self.assertTrue(int(pop.get_x()[0][-3]) == pop.get_x()[0][-3])
+        self.assertTrue(int(pop.get_x()[0][0]) != pop.get_x()[0][0])
+        self.assertTrue(int(pop.get_x()[0][1]) != pop.get_x()[0][1])
+
+
+class random_decision_vector_test_case(_ut.TestCase):
+    """Test case for random_decision_vector
+
+    """
+
+    def runTest(self):
+        from .core import random_decision_vector, set_global_rng_seed
+        set_global_rng_seed(42)
+        x = random_decision_vector(lb = [1.1,2.1,-3], ub = [2.1, 3.4,5], nix = 1)
+        self.assertTrue(int(x[-1]) == x[-1])
+        self.assertTrue(int(x[1]) != x[1])
+        set_global_rng_seed(42)
+        y = random_decision_vector(lb = [1.1,2.1,-3], ub = [2.1, 3.4,5], nix = 1)
+        self.assertTrue((x == y).all())
```

Comment:
those are tested in the C++ part. I usually only test the boost interface in test.py